reconcile/ocm/types.py

```diff
@@ -55,5 +55,9 @@ class OCMSpec(BaseModel):
 
     class Config:
         smart_union = True
-        # This is need to populate by either console_url or consoleUrl, for instance
-        allow_population_by_field_name = True
+        validate_assignment = True
```

Comment:
If you remove `allow_population_by_field_name`, then you can no longer set this via `server_url`. Is this the expected behavior? I believe that the `None` that you're getting here is because the alias `serverUrl` is missing, so the default value is used. It might make sense to add another test where `server_url` is set to a valid string.

There are at least a few places we're doing that like: https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/ocm.py#L179

Reply:
You were right. I had in mind the assignment was made using the object attribute instead of the constructor.
Added the option again.